man/tpm2_contextload.1.md

```diff
@@ -0,0 +1,53 @@
+% tpm2_contextload(1) tpm2-tools | General Commands Manual
+
+# NAME
+
+**tpm2_contextload**(1) - Loads the session into the TPM, so that it can be
+referenced using the TPM session handle.
+
+# SYNOPSIS
+
+**tpm2_contextload** [*OPTIONS*] [*ARGUMENT*]
+
+# DESCRIPTION
+
+**tpm2_contextload**(1) - Loads the session into the TPM, so that it can be
+referenced using the TPM session handle.
+
+The tool returns the TPM handle the session is currently loaded on, and that
+handle value can be used directly when communicating directly with the TPM.
+
+**NOTE**: The session context file will not be usable on subsequent calls. As
+all tools invoke contextload/contextsave before/after using the session, the
+contextload call for subsequent commands will fail as the session is loaded
+already.
+
+**NOTE 2**: When working with the in-kernel resource manager, the contextload
+command should be executed directly against the TPM (using the --tcti option)
+, as the in-kernel RM flushes the objects upon exiting.
+
+* **ARGUMENT** the session context file.
+
+## References
+
+[common options](common/options.md) collection of common options that provide
+information many users may expect.
+
+[common tcti options](common/tcti.md) collection of options used to configure
+the various known TCTI modules.
+
+# EXAMPLES
+
+## Create a policy session, contextload it into the TPM
+
+```bash
+
+tpm2 startauthsession -S session.ctx --policy-session
+
+tpm2 contextload -T "device:/dev/tpm0" session.ctx
+
+```
```

Comment:
Expand this example with your specific use case, so we can show an end-to-end scenario:
```bash
# use the session handle here for whatever things need it. In this case, we will set up the key for ....
your commands here
```

Reply:
Ok, there is an issue here though.

The end-to-end use-case doesn't quite work, because there are a few things that are slightly broken on the Kernel keyring trusted keys implementation for TPM2 (which was our original use case for this). It's unlikely that it will be fixed anytime soon, and when it gets fixed the subsystem may end up with a different interface to do these things.

So, at this point, I can't present a valid use-case for this tool. There may be others other people may have, but I'm not sure you'd want to keep the tool just in case?

Reply:
> Ok, there is an issue here though.
> 
> The end-to-end use-case doesn't quite work, because there are a few things that are slightly broken on the Kernel keyring trusted keys implementation for TPM2 (which was our original use case for this). It's unlikely that it will be fixed anytime soon, and when it gets fixed the subsystem may end up with a different interface to do these things.
> 
> So, at this point, I can't present a valid use-case for this tool. There may be others other people may have, but I'm not sure you'd want to keep the tool just in case?

uhh, that sucks. The beautiful thing about PRs is that we can keep em around. Just don't kill your branch off. We can close this, but if the kernel stuff gets fixed and we have a full end-to-end need, I have no issues with merging this.

Reply:
Ok, I'll close this PR for now, not to have it dangling around here, but will leave the code if we later on end up being able to get the kernel side working.

Thanks for your help!